examples/test/test_qnn_tooklit.py

```diff
@@ -11,35 +13,54 @@
 from olive.common.utils import retry_func, run_subprocess
 
 
-def download_qnn_sdk():
+def download_qnn_sdk(target_path=None):
     """Download the qnn sdk."""
+    blob, download_path = "", ""
+    if platform.system() == "Windows":
+        blob, download_path = "qnn_sdk_windows.zip", "qnn_sdk_windows.zip"
+    elif platform.system() == "Linux":
+        blob, download_path = "qnn_sdk_linux.zip", "qnn_sdk_linux.zip"
+
     download_azure_blob(
         container="olivetest",
-        blob="qnn_sdk_linux.zip",
-        download_path="qnn_sdk_linux.zip",
+        blob=blob,
+        download_path=download_path,
     )
-    target_path = Path().resolve()
-    run_subprocess(cmd=f"unzip qnn_sdk_linux.zip -d {str(target_path)}", check=True)
+    if not target_path:
+        target_path = Path().resolve() / "qnn_sdk"
+    target_path.mkdir(parents=True, exist_ok=True)
+    if platform.system() == "Windows":
+        cmd = f"powershell Expand-Archive -Path {download_path} -DestinationPath {str(target_path)}"
+        run_subprocess(cmd=cmd, check=True)
+    elif platform.system() == "Linux":
+        run_subprocess(cmd=f"unzip {download_path} -d {str(target_path)}", check=True)
     return target_path
 
 
 @pytest.fixture(scope="module", autouse=True)
 def setup():
     """Setups any state specific to the execution of the given module."""
+    tmp_dir = tempfile.TemporaryDirectory()
+    tmp_path = Path(tmp_dir.name)
```

Comment:
I suppose [https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/consider-using-with.html](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/consider-using-with.html) will be raised?

Reply:
that's the reason I suggest to use [https://docs.pytest.org/en/7.4.x/how-to/tmp_path.html](https://docs.pytest.org/en/7.4.x/how-to/tmp_path.html)